hsds/datanode.py

```diff
@@ -165,8 +167,16 @@ async def bucketScan(app):
                 tb = traceback.format_exc()
                 print("traceback:", tb)
 
-        log.info(f"bucketScan - sleep: {async_sleep_time}")
-        await asyncio.sleep(async_sleep_time)
+            last_action = time.time()
+
+        now = time.time()
+        if (now - last_action) > async_sleep_time:
```

Comment:
If `last_action` is set to the current time before the scanning loop is performed/at the end of each scan loop, won't this always be true?

Reply:
If there are no root_ids, last_action doesn't get updated, so eventually the if case will get executed.